packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -3,102 +3,202 @@
+    annotations: FluidErrorAnnotations = {},
+): asserts errorObject is IFluidErrorBase {
+    // If errorObject isn't suitable to annotate as IFluidErrorBase, bail.
+    assert(isRegularObject(errorObject) && !Object.isFrozen(errorObject),
+        "Cannot annotate a non-object or frozen error");
+
+    // Patch in any missing properties of IFluidErrorBase
+    const errorTypeIfNone = isErrorLike(errorObject)
+        ? errorObject.name
+        : typeof(errorObject);
+    const fluidError = patchFluidErrorBuilder(
+        errorObject as FluidErrorBuilder,
+        prepareFluidErrorTemplate(errorObject, errorTypeIfNone, annotations.errorCodeIfNone),
+    );
 
-/** Type guard for RwLoggingError.  Not exported. */
-const isRwLoggingError = (x: any): x is RwLoggingError =>
-    typeof x?.addTelemetryProperties === "function" && isILoggingError(x);
+    mixinTelemetryPropsWithFluidError(fluidError, annotations.props);
+}
 
 /**
- * Annotate the given error object with the given logging props
- * @returns The same error object passed in if possible, with telemetry props functionality mixed in
+ * Helper type, not exported.
+ * Makes IFluidErrorBuilder's props mutable and optional for building one up on an existing object
  */
-export function annotateError(
-    error: unknown,
-    props: ITelemetryProperties,
-): ILoggingError {
-    if (isRwLoggingError(error)) {
-        error.addTelemetryProperties(props);
-        return error;
-    }
+type FluidErrorBuilder = {
+    -readonly [P in keyof IFluidErrorBase]?: IFluidErrorBase[P];
+};
+
+/**
+ * Helper type, not exported.
+ * Makes IFluidErrorBuilder's props all type unknown for querying if they're already valid
+ */
+ type FluidErrorProperties = {
+    -readonly [P in keyof IFluidErrorBase]: unknown;
+};
 
-    if (isRegularObject(error)) {
-        // Even though it's not exposed, fully implement IRwLoggingError for subsequent calls to annotateError
-        const loggingError = error as RwLoggingError;
+/** Returns a template IFluidErrorBase with values based on the inputs provided, for use with patchFluidErrorBuilder */
+function prepareFluidErrorTemplate(
+    originalError: unknown,
+    errorTypeIfNone: string,
+    errorCodeIfNone: string | undefined,
+): IFluidErrorBase {
+    const fullErrorTypeIfNone = `none (${errorTypeIfNone})`;
+    const fullErrorCodeIfNone = errorCodeIfNone === undefined
+        ? "none"
+        : `none (${errorCodeIfNone})`;
+    const { errorType, fluidErrorCode, message, name, stack } = originalError as FluidErrorProperties;
+    return {
+        errorType:
+            typeof errorType === "string"
+                ? errorType
+                : fullErrorTypeIfNone,
+        fluidErrorCode:
+            typeof fluidErrorCode === "string"
+                ? fluidErrorCode
+                : fullErrorCodeIfNone,
+        message:
+            typeof message === "string" || message === undefined
+                ? undefined  // Don't patch - it's already valid
```

Comment:
If it was a string, it would be come undefined - this looks wrong

Reply:
Same thing below.

Reply:
Yeah, I felt this was a little weird - `undefined` for these means do nothing in the other function.  But that's coupling logic at a distance, not great...

Honestly, one big reason is code I saw from you that was concerned with whether setting `.stack` on an `Error` would throw.  So I'm opting here for doing instead of overwriting.